### PR TITLE
Added NNCF quantization for Distil-Whisper model

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
In total performance boost is about 1.75x with &lt; 1% acc. drop.</p><figure class="table">

Model | Size | Number of calls on 50 samples | Inference time on 50 samples | Performance boost | Accuracy on 50 samples
-- | -- | -- | -- | -- | --
Original encoder | 1244 MB | 50 | 76.24 sec. | - | -
Original decoder | 233 MB | 50 | 1.035 sec. | - | -
Original decoder with past | 221 MB | 1440 | 15.53 sec. | - | -
Original whole pipeline | - | 50 | 94.84 sec. | - | 82.38 %
Quantized encoder | 626 MB | 50 | 54.32 sec. | 1.4x | -
Quantized decoder with past | 111 MB | 1397 | 7.86 sec. | 1.98x | -
Quantized whole pipeline | - | 50 | 54.32 sec. | 1.75x | 82.17 %

First step decoder is not quantized as it's inference time contribution is negligible.

Ticket 124423

### CODE REVIEW

File: notebooks/267-distil-whisper-asr/267-distil-whisper-asr.ipynb [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "\n",
```
Comment (l-bat) [inline notebooks/267-distil-whisper-asr/267-distil-whisper-asr.ipynb:15]: <blockquote> Quantization is the simplest scenario</blockquote><p> </p><p><a href="https://github.com/openvinotoolkit/openvino_notebooks/pull/1184#discussion_r1334286561" target="_blank">https://github.com/openvinotoolkit/openvino_notebooks/pull/1184#discussion_r1334286561</a></p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1439/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (l-bat) [inline notebooks/267-distil-whisper-asr/267-distil-whisper-asr.ipynb:15]: <p>Let's compare the transcription results for original <strong>and </strong>quantized models.</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1439/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (nikita-savelyevv) [reply notebooks/267-distil-whisper-asr/267-distil-whisper-asr.ipynb:15]: Fixed

Comment (nikita-savelyevv) [reply notebooks/267-distil-whisper-asr/267-distil-whisper-asr.ipynb:15]: Removed

**Review by MaximProshin — APPROVED** [review]
LGTM

**Review by l-bat — APPROVED** [review]

**Review by eaidova — APPROVED** [review]

**Pull request merged**